### PR TITLE
Feature/qc 5 refactor with hooks

### PR DESCRIPTION
**Pull request by Shushko**

### CODE REVIEW

File: src/components/ChatSection/ChatWindow/ChatItem/MessageItem/MessageItem.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import classes from "./MessageItem.module.sass";
+import moment from "moment";
+
+const MessageItem = ({ messageItem, currentUserId }) => {
```
Comment (lolaignatova) [inline src/components/ChatSection/ChatWindow/ChatItem/MessageItem/MessageItem.js:5]: you can simplify it like that
const MessageItem = ({ messageItem, currentUserId }) => ({
  // put here jsx
});

File: src/components/ChatSection/ChatWindow/ChatItem/MessageItem/MessageItem.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return (
+        <div className={ classes.message_item }>
+            <span className={ classes.message_content }>{ messageItem.message }</span>
+            <div>
+                { messageItem.userId === currentUserId && (!messageItem.isDelivered || !messageItem.isRead) &&
```
Comment (lolaignatova) [inline src/components/ChatSection/ChatWindow/ChatItem/MessageItem/MessageItem.js:10]: can you please simplify it

File: src/components/ChatSection/ChatWindow/ChatWindow.js [hunk 3]
```diff
@@ -1,141 +1,124 @@
...
 
-    render() {
-        return (
-            <div className={ classes.chats_window }>
-                {
-                    !this.props.match.params.dialogId ?
-                        <span className={ classes.start_text }>Select a dialog and start communication</span> :
+const mapStateToProps = (state) => ({
+    dialogsState: state.dialogsDataReducer,
+    currentUser: state.dialogsDataReducer.currentUser,
```
Comment (lolaignatova) [inline src/components/ChatSection/ChatWindow/ChatWindow.js:112]: you can pass to the props whole dialogsDaaReducer. It contains currentUser in it

File: src/components/ChatSection/SendMessagePanel/SendMessagePanel.js [hunk 4]
```diff
@@ -3,15 +3,15 @@ import classes from './SendMessagePanel.module.sass';
...
 
-const SendMessagePanel = (props) => {
+const SendMessagePanel = ({ currentUser, interlocutor, currentDialog, onSubmit }) => {
     return (
         <div
-            className={ props.currentDialog ? classes.send_message_panel_wrap : classes.send_message_panel_wrap_disable }>
+            className={ currentDialog ? classes.send_message_panel_wrap : classes.send_message_panel_wrap_disable }>
             <div className={ classes.send_message_panel }>
-                <img src={ props.currentUser.avatar } alt="ava"/>
+                <img src={ currentUser.avatar } alt="ava"/>
```
Comment (lolaignatova) [inline src/components/ChatSection/SendMessagePanel/SendMessagePanel.js:11]: what does it mean "ava"?

File: src/components/ChatSection/SendMessagePanel/SendMessagePanelContainer.js [hunk 5]
```diff
@@ -1,46 +1,42 @@
...
-            const interlocutor = dialog && this.findInterlocutor(dialog)
-            this.setState({ dialog, interlocutor })
-        }
-    }
+    const findInterlocutor = (dialog) => dialog.members.find(m => m.id !== dialogsData.currentUser.id);
 
-    findInterlocutor = (dialog) => dialog.members.find(m => m.id !== this.props.dialogsData.currentUser.id)
+    const onSubmit = (formData) => {
+        addNewMessage(
+            interlocutor.id,
```
Comment (lolaignatova) [inline src/components/ChatSection/SendMessagePanel/SendMessagePanelContainer.js:22]: call addNewMessage from child please

File: src/components/DialogsSection/UserDialogs.js [hunk 6]
```diff
@@ -1,71 +1,78 @@
...
-                                }
-                            ) }
-                        </div>
-                    </div> }
-            </>
-        )
-    }
+    return (
+        <>
+            { !dialogsData.currentUser ? <div/> :
```
Comment (lolaignatova) [inline src/components/DialogsSection/UserDialogs.js:47]: please change it with dialogsData.currentUser &&

File: src/components/FindUser/FindUser.js [hunk 7]
```diff
@@ -11,41 +11,41 @@ import { createNewDialog, onChangeCurrentDialog } from "../../redux/dialogsData/
...
         }
     }
 
-    getUserItems = () => {
-        if (this.props.foundUsers.length) {
-            return this.props.foundUsers.map(item => item.id === this.props.currentUser.id ? '' :
-                <UserItem user={ item } addNewDialog={ this.addNewDialog } key={ item.id } />)
+    const getUserItems = () => {
+        if (props.foundUsers.length) {
+            return props.foundUsers.map(item => item.id === props.currentUser.id ? '' :
```
Comment (lolaignatova) [inline src/components/FindUser/FindUser.js:38]: so returned array can look like this ['', <UserItem/>, '', '']? Is it correct? Maybe better will be filter an array?

File: src/common/StartText.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+const StartText = () => {
+    const startTextWrap = {
```
[the remark this answers is not in the document]
Comment (Shushko) [reply src/common/StartText.js:4]: It's so small component, that I didn't want to create a new directory with style file and js file.

File: src/components/ChatSection/ChatWindow/ChatWindow.js [hunk 9]
```diff
@@ -1,141 +1,127 @@
...
+        currentChat.current = dialogsState.dialogs.find(d => d.dialogId === dialogsState.currentDialog);
+        prevMessageUserId.current = null;
+        const chatMessages = currentChat.current.messages;
+        const chatMembers = currentChat.current.members;
+        const lastReadMessage = getLastReadMessage();
+        const lastMessage = chatMessages[chatMessages.length - 1];
+        const interlocutor = chatMembers.find(member => member.id !== currentUser.id);
+        if (!lastReadMessage) {
+            lastReadMessageRef !== chatStart.current && setLastReadMessageRef(chatStart.current)
+            // lastReadMessageRef.current = chatStart.current;
```
Comment (lolaignatova) [inline src/components/ChatSection/ChatWindow/ChatWindow.js:71]: please, remove commended code

**Review by lolaignatova — APPROVED** [review]

**Pull request merged**